extensions/mssql/src/features.ts

```diff
@@ -43,7 +43,12 @@ export class AccountFeature implements StaticFeature {
 		let timeToLiveInSeconds = 10;
 		this.tokenCache = new DataItemCache(this.getToken, timeToLiveInSeconds);
 		this._client.onRequest(contracts.SecurityTokenRequest.type, async (request): Promise<contracts.RequestSecurityTokenResponse | undefined> => {
-			return this.tokenCache.getData(request);
+			return await this.tokenCache.getData(request);
+		});
+		this._client.onNotification(contracts.RefreshTokenNotification.type, async (request) => {
+			// send refresh token to sts to get updated
```

Comment:
`Refresh token, then inform client the token has been updated. This is done as separate notification messages due to the synchronous processing nature of STS currently <provide link to relevant issue>`